xarray/backends/netCDF4_.py

```diff
@@ -198,24 +201,40 @@ class NetCDF4DataStore(WritableCFDataStore, DataStorePickleMixin):
 
     This store supports NetCDF3, NetCDF4 and OpenDAP datasets.
     """
-    def __init__(self, filename, mode='r', format='NETCDF4', group=None,
+    def __init__(self, netcdf4_dataset, mode='r', group=None, writer=None,
+                 opener=None, autoclose=False):
+        self.ds = _get_netcdf4_group(netcdf4_dataset, group)
+
+        if autoclose and opener is None:
+            raise ValueError('autoclose requires an opener')
+
+        self._mode = mode
+        self._autoclose = autoclose
+        self._opener = opener
+        self._isopen = True
+        self.format = self.ds.data_model
+        self._filename = self.ds.filepath()
+        self.is_remote = is_remote_uri(self._filename)
+        super(NetCDF4DataStore, self).__init__(writer)
+
+    @classmethod
+    def open(cls, filename, mode='r', format='NETCDF4', group=None,
                  writer=None, clobber=True, diskless=False, persist=False,
                  autoclose=False):
+        import netCDF4 as nc4
+
         if format is None:
             format = 'NETCDF4'
-        opener = functools.partial(_open_netcdf4_group, filename, mode=mode,
+        opener = functools.partial(_open_netcdf4_group, filename,
```

Comment:
I was looking in `common.py` and it seems like mode gets overridden and thus was not really needed:
https://github.com/pydata/xarray/blob/8e541deca2e20efe080aa1bca566d9966ea2f244/xarray/backends/common.py#L255-L265

Happy to defer to your judgement.